### PR TITLE
OE-136: Added task aggregator

### PR DESCRIPTION
**Pull request by singhraman4282**
**What's this do?**
Adds a task aggregator 

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/OE-136

**How should this be tested? / Do these changes have associated tests?**
Can not be QA tested

**Dependencies for merging? Releasing to production?**
N/A

**Does this include changes that require a new SimplyE build for QA?**
Not yet

**Has the application documentation been updated for these changes?**
N/A

**Did someone actually run this code to verify it works?**
Raman Singh verified it

### CODE REVIEW

**Comment by singhraman4282** [comment]
@ettore @ErnestFan I am not quite sure if this PR makes sense since Apple announced await async functionality and actors in the latest WWDC and this PR is compensating for the lack of async await methods. Once we update to Swift 5.5, we can utilize async await and the code for Axix book download will be MUCH cleaner.

**Comment by singhraman4282** [comment]
Apart from being massive, it also feels like this is throw away work

File: Simplified/AxisService/NYPLAxisItemDownloader.swift [hunk 1]
```diff
@@ -11,14 +11,17 @@ import Foundation
 protocol NYPLAxisDownloadProgressListening: class {
   func downloaderDidTerminate()
   func downloadProgressDidUpdate(_ progress: Double)
+  func didFinishAllDownloads()
 }
 
 protocol NYPLAxisItemDownloading: class {
   var delegate: NYPLAxisDownloadProgressListening? { get set }
-  var dispatchGroup: DispatchGroup { get }
+  var tasksSynchnorizer: NYPLAxisTasksSynchorizing { get }
```
Comment (ErnestFan) [inline Simplified/AxisService/NYPLAxisItemDownloader.swift:19]: minor: mis-spell - synchronizer and synchronizing

File: Simplified/AxisService/NYPLAxisLicenseService.swift [hunk 2]
```diff
@@ -190,22 +189,21 @@ class NYPLAxisLicenseService: NYPLAxisLicenseHandling {
...
     }
     
     return Data(base64Encoded: encryptedContentValue)
   }
   
-  func logLicenseErrorAndLeave(summary: String,
-                               reason: String,
-                               additionalInfo: [String: String] = [:]) {
+  func logLicenseErrorAndTerminateDownload(
+    summary: String, reason: String, additionalInfo: [String: String] = [:]) {
```
Comment (ErnestFan) [inline Simplified/AxisService/NYPLAxisLicenseService.swift:206]: I think it's easier to read with the indentation like before (each parameter on a new line), just a suggestion

File: Simplified/AxisService/NYPLAxisTasksSynchnorizer.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+extension NYPLAxisTasksSynchorizing {
+  func startSynchronizedTaskWhenIdle() {
+    waitForSynchronizedTaskToFinish()
+    startSynchronizedTask()
+  }
+}
+
+/// Wrapper class for aggregating a set of tasks and synchronizing behavior
+class NYPLAxisTasksSynchnorizer: NYPLAxisTasksSynchorizing {
```
Comment (ettore) [inline Simplified/AxisService/NYPLAxisTasksSynchnorizer.swift:27]: so, on a deeper look, this class is actually kind of synonym of DispatchGroup. It wraps it with equivalent apis. Did you add it mainly to add more appropriate nouns and verbs that match the objective better?

File: SimplifiedTests/NYPLAxisTests/NYPLAxisItemDownloaderTests.swift [hunk 4]
```diff
@@ -28,14 +29,10 @@ class NYPLAxisItemDownloaderTests: XCTestCase {
   func testDownloaderShouldNotContinueOnFailedDownload() {
     let stopExpectation = self.expectation(
       description: "Downloader should not continue after download failure")
-    let leaveExpecation = self.expectation(
-      description: "Downloader should leave dispatch group on failed download")
+    let notifyExpecation = self.expectation(
```
Comment (ettore) [inline SimplifiedTests/NYPLAxisTests/NYPLAxisItemDownloaderTests.swift:32]: typo

**Pull request closed without merging**